### ISSUE TITLE
Be strict about type Hint and use Mypy properly, check if possible to do diff to see new type hint issue.

### ISSUE THREAD

**Issue by inkhey** [tracim/tracim#1588]
## Bug description & symptom

Currently we do not rely on type hint with mypy because Tracim has a big code base and too many mypy errors.

## Expected

- [ ] Find a solution to return only diff of mypy error in order to see new typing issue, and use it systematically when writing code.
- [ ] Fix old code source-base typing issue.

**Comment by buxx** [comment]
Note: expected " Find a solution to return only diff of mypy error" can be done before second.

**Comment by inkhey** [comment]
> Note: expected " Find a solution to return only diff of mypy error" can be done before second.
 
Yes, it **should** be done before second, because as soon we fix typing issue in whole code, there is no more need to have diff.

**Comment by lebouquetin** [comment]
The nice way to gradually improve would be to allow only a decreasing number of errors (like what travis-ci does with code coverage)
